### ISSUE TITLE
How is Grad-CAM picture drawed?

### ISSUE THREAD

**Issue by machengcheng2016** [BeierZhu/Prompt-align#2]
Really solid work and really admire you! 
I've got a simple question. How are the Grad-CAM pictures drawed in Figure. 2? Specifically, in Grad-CAM, which target layer do you choose?

**Comment by BeierZhu** [comment]
Dear Chengcheng Ma:

Thank you for your interest in ProGrad.

For Figure 2, we use the last convolutional layer of CLIP-ResNet50 (clip_model.visual.layer4[2]) to draw the Grad-CAM.
And adapt the code from this link [Grad-CAM](https://zhuanlan.zhihu.com/p/453689298)

Best

**Comment by machengcheng2016** [comment]
Thanks for your reply.
I think the clip_model.visual.layer4[2] always remain fixed, because both CoOp and ProGrad only train the prompt about the text model and never change the parameters of visual model. So why are the GradCAM pictures different?

**Comment by BeierZhu** [comment]
Because the classification head generated by prompt changes

**Comment by BeierZhu** [comment]
CAM activation depends on the derivative of the logits which is a function of classification head

**Comment by machengcheng2016** [comment]
Understood. Much appreciate!
